crates/rethnet_eth/src/transaction/signed/legacy.rs

```diff
@@ -92,3 +77,67 @@ impl rlp::Decodable for LegacySignedTransaction {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use revm_primitives::Address;
+    use secp256k1::SecretKey;
+
+    use super::*;
+
+    fn dummy_request() -> LegacyTransactionRequest {
+        let to = Address::from_str("0xc014ba5ec014ba5ec014ba5ec014ba5ec014ba5e").unwrap();
+        let input = hex::decode("1234").unwrap();
+        LegacyTransactionRequest {
+            nonce: 1,
+            gas_price: U256::from(2),
+            gas_limit: 3,
+            kind: TransactionKind::Call(to),
+            value: U256::from(4),
+            input: Bytes::from(input),
+        }
+    }
+
+    fn dummy_private_key() -> SecretKey {
+        SecretKey::from_str("e331b6d69882b4cb4ea581d88e0b604039a3de5967688d3dcffdd2270c0fd109")
+            .unwrap()
+    }
+
+    #[test]
+    fn test_legacy_signed_transaction_encoding() {
```

Comment:
Just informational: I was a bit surprised to find the signature testing methods in the tests for the transaction structs as signing is implemented in the transaction request structs.